client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -233,99 +207,136 @@ def predict(
 
         """
-        num_retries = 0
-        while num_retries < max_retries:
-            response = self._predict(x, compress)
-
-            if response.ok:
-                return _utils.body_to_json(response)
-            elif response.status_code in (
-                400,
-                502,
-            ):  # possibly error from the model back end
-                try:
-                    data = _utils.body_to_json(response)
-                except ValueError:  # not JSON response; 502 not from model back end
-                    pass
-                else:  # from model back end; contains message (maybe)
-                    # try to directly print message, otherwise line breaks appear as '\n'
-                    msg = data.get("message") or json.dumps(data)
-                    raise RuntimeError(
-                        "deployed model encountered an error: {}".format(msg)
-                    )
-            elif not (
-                response.status_code >= 500 or response.status_code in (404, 429)
-            ):  # clientside error
-                break
-
-            sleep = 0.3 * (2 ** (num_retries + 1))
-            print(
-                "received status {}; retrying in {:.1f}s".format(
-                    response.status_code, sleep
-                )
+        if always_retry_404 or always_retry_429:
+            warnings.warn(
+                "Deprecation warning: The `always_retry_404` and `always_retry_429` parameters are deprecated and "
+                "will removed in an upcoming version;  Status codes that should be retried are passed in the "
+                "`retry_status` parameter which includes 404 and 429 by default.",
+                category=FutureWarning,
             )
-            time.sleep(sleep)
-            if (response.status_code == 404 and always_retry_404) or (  # model warm-up
-                response.status_code == 429 and always_retry_429
-            ):  # too many requests
-                num_retries = min(num_retries + 1, max_retries - 1)
-            else:
-                num_retries += 1
+        # Update retry config and re-mount it to the existing session if it differs from the current one.
+        self._session = http_session_utils.alter_retry_config(self._session,
+                                                              max_retries=max_retries,
+                                                              status_forcelist=retry_status,
+                                                              backoff_factor=backoff_factor)
+
+        if prediction_id:
+            http_session_utils.update_session_headers(self._session, {'verta-request-id': prediction_id})
+            response = self._predict(x, compress)
+            http_session_utils.update_session_headers(self._session, {'verta-request-id': prediction_id}, remove=True)
+        else:
+            response = self._predict(x, compress)
+        return _utils.body_to_json(response)
```

Comment:
Should we just call `predict` and drop the first returned value? The implementation is about hte same

Reply:
Ah, so are you suggesting that all this logic should be in `predict_with_id` and `predict` should just call that and drop the id to return only the results?  I'm not sure why I didn't think of that!  I think I just had the inverse relationship in my head.  I agree that would be much simpler

Reply:
yeah. the logic is already in `predict_with_id` from what I could tell